### ISSUE TITLE
Completion Suggester w/ context filter doesn't work with UTF8 characters

### ISSUE THREAD

**Issue by lyrixderaven** [elastic/elasticsearch#20509]
**Elasticsearch version**: 2.3.3

**OS version**: Ubuntu

When using the completion suggester, it seems the categories can not contain UTF8-Characters, such as Umlauts; if so, the suggest query doesn't return anything.

For instance, the following indexed document contains an umlaut in the context ""MyUmlautSite\u00e4":

```
{
    "_index": "test",
    "_type": "document",
    "_id": "1",
    "_score": 0.59573317,
    "_source": {
        "suggest_completion": {
            "input": [
                "mobilit\u00e4tssysteme",
                "technologien",
                "l\u00f6sungsans\u00e4tzen",
                "integration",
                "personenmobilit\u00e4t"
            ],
            "context": {
                "sites": [
                    "all",
                    "MyUmlautSite\u00e4"
                ]                            }
        },
        "Title": "TestTitle",
        "sites": [
            "MyUmlautSite\u00e4"
        ]
    }
}
```

The umlaut's representation as \u00e4 stems from the json formatting, we confirmed it's actually 'ä' in the index as expected.

When requesting suggestions like this, no results are being returned:

```
{
        "index": "test",
        "size": 0,
        "body": {
            "suggest": {
                "suggest": {
                    "text": "integr",
                    "completion": {
                        "field": "suggest_completion",
                        "size": 10,
                        "context": {
                            "sites": "MyUmlautSite\u00e4"
                        }
                    }
                }
            }
        }
    },
``` 

Again, the actual request is being done with true unicode encoded Umlauts.

Any ideas? Known bug?

**Comment by clintongormley** [comment]
@areek if you haven't done so already, could you add tests to be sure this works in 5.0?

**Comment by areek** [comment]
@lyrixderaven this is a known bug for context filtering in completion suggester in 2.x. This works as expected in 5.0.
@clintongormley I manually tested this with 5.0 completion suggester, will also add a test for this

**Comment by areek** [comment]
I added test for this in [5.0] (https://github.com/elastic/elasticsearch/commit/c8843d6c7dd6101afa6b4aa9d1842e52519fd05b), [5.x] (https://github.com/elastic/elasticsearch/commit/a043ac5860ccaf6938f3e60b8f9ce01a3885f2ae) and [master] (https://github.com/elastic/elasticsearch/commit/16ed2fb423fd387faa513cbed991a9fcb64bc572
). Closing as resolved in 5.0
